Sources/Utility/Toolchain.swift

```diff
@@ -0,0 +1,58 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+ -------------------------------------------------------------------------
+ This file defines support functions for resources tied to the installed
+ location of the containing product binary.
+ */
+
+import func POSIX.getenv
+import func POSIX.popen
+
+public protocol Installation {
+    static func which(cmd: String) -> String
+}
+
+public struct Toolchain: Installation {
+    public static func which(cmd: String) -> String {
+        if let exepath = Process.arguments.first {
+            if let path = try? Path.join(exepath, "..", cmd).abspath() where path.isFile {
+                return path
+            }
+        }
+
+    #if os(OSX)
+        if let cmdpath = (try? popen(["xcrun", "--find", cmd]))?.chuzzle() {
+            return cmdpath
+        }
+    #endif
+        
+        return cmd
+    }
+
+    //TODO better
+    public static var prefix: String {
+        return swiftc.parentDirectory.parentDirectory.parentDirectory
+    }
+
+    /// the location of swiftc relative to our installation
+    public static let swiftc = getenv("SWIFT_EXEC") ?? Toolchain.which("swiftc")
```

Comment:
Seems like your environment is not setup correctly if xcrun is giving you the wrong `swiftc`. /cc @ddunbar